content/Watsonx/CodeAssistant/102.md

```diff
@@ -1,117 +1,167 @@
+1. To see this in action, open the Ansible Playbook in the following directory: `lightspeed-demos-main/playbooks/cloud/aws/demo_provision_ec2_instance.yml`. Examine **Line 26** and **Line 33**. **Task 1** (**Line 26**) and **Task 2** (**Line 33**) are responsible for gathering information about a network subnet that is to be provisioned and then creating a virtual private cloud (VPC) definition based on those details.
 
-    The actions that you take when a recommendation is provided impact the training process of the model.
+2. Generate an AI-suggested code block for **Task 1** (**Line 26**) by un-commenting the line, placing the cursor at the end of the line, pressing `RETURN`, and then confirming with `TAB`. Afterwards, generate an AI-suggested code block for **Task 2** but do **not** press `TAB` to confirm the code block.
 
-    If a recommendation is accepted, and then further edits are performed, then the act of changing the recommendation to something else will be considered a modification of the recommendation. This will tell Ansible and IBM watsonx Code Assistant that the recommendation required extra editing in order to meet the intended use. This information will be used for context in training the model for similar prompts in the future.
+    The first round of AI-generated code for **Task 1** produces a code block with a `register: subnet_info` line — the result of which is to assign this VPC definition to a variable `subnet_info`.
 
+    The AI-generated code that follows for **Task 2** recommends a code block with `vpc_subnet_id: "{ { subnet_info.subnets[0].subnet_id } }"` as the value associated with the VPC's subnet ID. Critically, the variable `subnet_info` that was generated in the previous Task is also referenced in the second Task. This demonstrates the contextual awareness of WCA in action.
 
+  ![](./images/task-variable-info.png)
 
+3. Now change the name of the variable that was generated from **Task 1** from `subnet_info` to `subnet_name`. Observe how the **Task 2** block is altered after deleting the previously-generated **Task 2** content and re-generate the code block by pressing `RETURN` at the end of the line for **Task 2**.
 
+  You will immediately notice that the new block of task code references the variable `subnet_name` that was modified just a moment ago. Once again, WCA has generated code suggestions that take into account the altered context and variables of the Playbook.
 
+  ![](./images/task-variable-name.png)
 
+## IBM watsonx Code Assistant for Red Hat Ansible Lightspeed L3 Course Quiz
+Congratulations! You have completed the necessary labs needed to complete the IBM watsonx Code Assistant for Red Hat Ansible Lightspeed L3 course quiz. The quiz can be found [here](https://learn.ibm.com/course/view.php?id=15649). **Note:** You do not need to watch the videos as the labs have covered all course content except for the quiz. You can click through the lessons and take the quiz.
```

Comment:
Could you add quiz alerts throughout the lab so that attendees can pay extra attention to parts of the lab related to quiz questions?

Reply:
The quiz questions [here](https://learn.ibm.com/course/view.php?id=15649) all want the user to run additional tasks in the playbooks that are outside the lab and ask about the responses from the AI-generated code

Reply:
Per the discussion in our meeting today, I added the quiz alerts.